src/openloco/windows/construction/ConstructionTab.cpp

```diff
@@ -217,6 +409,1324 @@ namespace openloco::ui::windows::construction::construction
+                id = 8;
+                break;
+            }
+
+            case TrackPiece::right_hand_curve_large: // loc_4A0705
+            {
+                if (gradient != TrackGradient::level)
+                    return std::nullopt;
+                id = 11;
+                if (rotation >= 12)
+                    break;
+                if (rotation >= 4)
+                    return std::nullopt;
+                id = 9;
+                break;
+            }
+
+            case TrackPiece::s_bend_left: // loc_4A0729
+            {
+                if (gradient != TrackGradient::level)
+                    return std::nullopt;
+                if (rotation >= 12)
+                    return std::nullopt;
+                id = 33;
+                if (rotation >= 8)
+                    break;
+                if (rotation >= 4)
+                    return std::nullopt;
+                id = 12;
+                break;
+            }
+
+            case TrackPiece::s_bend_right: // loc_4A0756
+            {
+                if (gradient != TrackGradient::level)
+                    return std::nullopt;
+                if (rotation >= 8)
+                    return std::nullopt;
+                id = 32;
+                if (rotation >= 4)
+                    break;
+                id = 13;
+                break;
+            }
+
+            case TrackPiece::s_bend_to_dual_track: // loc_4A077C
+            {
+                if (gradient != TrackGradient::level)
+                    return std::nullopt;
+                if (rotation >= 8)
+                    return std::nullopt;
+                id = 40;
+                if (rotation >= 4)
+                    break;
+                id = 38;
+                break;
+            }
+
+            case TrackPiece::s_bend_to_single_track: // loc_4A07A2
+            {
+                if (gradient != TrackGradient::level)
+                    return std::nullopt;
+                if (rotation >= 12)
+                    return std::nullopt;
+                id = 41;
+                if (rotation >= 8)
+                    break;
+                if (rotation >= 4)
+                    return std::nullopt;
+                id = 39;
+                break;
+            }
+
+            case TrackPiece::turnaround: // loc_4A07C0
+            {
+                if (gradient != TrackGradient::level)
+                    return std::nullopt;
+                if (rotation >= 12)
+                    return std::nullopt;
+                id = 43;
+                if (rotation >= 8)
+                    break;
+                id = 42;
+                if (rotation >= 4)
+                    return std::nullopt;
+                break;
+            }
+        }
+
+        if (rotation < 12)
+            rotation &= 3;
+
+        return TrackPieceId{ id, rotation };
+    }
+
+    // 0x00488B4D
+    static void refreshSignalList(uint8_t* signalList, uint8_t trackType)
```

Comment:
Move to object manager?

Reply:
No we shouldn't move these functions they are specific to the window. These are arrays for the window afaik